ex1/functions.py

```diff
@@ -65,23 +116,125 @@ def f():
 # compare against. The function should return a list with equal length as the
 # input list containing for each number in the original list either True or
 # False if the number was greater than or equal to n.
+def check_list(list_of_numbers, n):
+	"""
+	Parameters
+	----------
+	list_of_numbers : list of int
+	n : int
+
+	Returns
+	-------
+	list of bool
+			True for every element in list_of_numbers greater than n, else False
+	
+    	"""
+	bool_list = [] # list with boolen values
+	for i in list_of_numbers:
+		if(i>=n):
+			bool_list.append(True)
+		else:
+			bool_list.append(False)
+
+	return bool_list
 
 # Define a function named check_list_nth that does the same as check_list but
 # uses every nth element of the input list (including the first one). You will
 # need a third input argument.
+def check_list_nth(list_of_numbers, n, tab):
+	"""
+	Parameters
+	----------
+	list_of_numbers : list of int
+	n, tab : int
+
+	Returns
+	-------
+	list of bool
+			True for tab-th element in list_of_numbers greater than n, else False
+	
+    	"""
+	bool_list = [] # list with boolen values
+	for i in (list_of_numbers[::tab]):
+		if(i>=n):
+			bool_list.append(True)
+		else:
+			bool_list.append(False)
+
+	return bool_list
 
 # Define a function named add_new_list that takes two inputs. A list l and a
 # second variable x to add to the list. Return a new list containing x as the
 # last element without modifying the original list.
+def add_new_list(org_list, x):
+	"""
+	Parameters
+	----------
+	org_list : list of optional
+	x : optional
+
+	Returns
+	-------
+	list of optional
+			New list was created, so the original list was not changed!
+    	"""
+	new_list = [] # list with optional values
+	for i in org_list:
+		new_list.append(i)
+	new_list.append(x)
+	return new_list
 
 # Define a function named remove_nth that takes a list and removes every nth
 # element (including the first one). Use a keyword named nth to set the default
 # value for nth to 2.
+def remove_nth(List, nth=2):
+	"""
+	Parameters
+	----------
+	List : list of optional
+	nth : int
+		default value 2
+
+	Returns
+	-------
+	list of optional
+			Original list was changed by removing every nth element!
```

Comment:
The Original list was not changed. But you return a new list. So this might be misleading.